### PR TITLE
[elm/en] Create intro to Elm

### PR DESCRIPTION
**Pull request by mgold**
Previously attempted at #1164 but this version is a bit more fleshed out, updated for 0.16, and I've paid special attention to the order in which concepts are introduced.

As cited in the other PR, there is no Markdown parser for Elm, so I used Haskell. It should be okay.

Feedback welcome, but the intention is to cover common cases and best practices, not be a definitive reference.

### CODE REVIEW

File: elm.html.markdown [hunk 1]
```diff
@@ -0,0 +1,346 @@
...
+-- Apply (call) a function by passing it arguments (no commas necessay).
+multiply 7 6 -- 42
+
+-- Partially apply a function by passing only some of its arguments.
+-- Then give that function a new name.
+double =
+  multiply 2
+
+-- Constants are similar, except there are no arguments.
+answer =
```
Comment (vendethiel) [inline elm.html.markdown]: anyway reason to have a newline here and before? Reads weirdly, but maybe the conventions in Elm are different from Haskell's

Comment (mgold) [inline elm.html.markdown]: That is indeed the convention, even though I don't like it, especially in this case (constants). I can ping some other Elm users if you want a third opinion.

Comment (vendethiel) [inline elm.html.markdown]: No, I'll trust you :). (and I'll just blame you if someone comes later and says it's wrong... :P)

**Comment by vendethiel** [comment]
Thanks!

**Comment by mgold** [comment]
Thanks for merging!

The Haskell syntax highlighting works fine. The only place it really screws up is the example of invalid code (I wouldn't object to removing it). It could stand to highlight Elm's port and alias keywords, but it's not a big deal.

**Pull request merged**